### DISCUSSION TITLE
S2 mini hardware diagnosis

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12326] [Hardware & Peripherals]
I've got a wemos S2 mini (https://www.wemos.cc/en/latest/_static/files/sch_s2_mini_v1.0.0.pdf) that won't give me a repl prompt out of the box, but the board is otherwise functional, I can load python into it. 

If I press both the rst(EN) & IO0 buttons then release rst then IO0 I can hear the comport make the windows connect noise. If I press & release rst by itself the comport makes the disconnect noise but does not reconnect. My first thought was to grab another S2 mini, but,. just my luck, the dud is the last one in the parts bin. 

At this point I figure it's got to be something simple. I check powering up with rst depressed. I figure if the C14 cap is dodgy, this trick will give me a USB connect when I release rst, no joy. I check the 10k pull up on IO18, it's functional. I give the board a good visual inspection with a USB microscope, nothing obvious missing or misaligned.

I've got this uneasy feeling that the symptoms above are signalling something obvious at me but I'm too dumb to see it. Any hardware gurus see something obvious or have a diagnosis I haven't tried?

**Comment by bidrohini** [comment]
I think you need to check your whole PCB with a multimeter. Check if there are any broken traces. Also, check if there is a short circuit somewhere. Match with the diagram/PCB layout and do the troubleshooting with your meter. 
By the way, if you decide to make your own Wemos D1 board, you can follow this design: 

https://www.pcbway.com/project/shareproject/Make_Your_own_Wemos_D1_Board_c1f7414d.html

**Comment by shariltumin** [comment]
Hey @kjm1102, it might be a firmware issue rather than hardware. In my experience, setting the correct USB flags in the "sdkconfig.board" file will decide whether you get REPL through USB or not.

These are my USB flags,
```
# USB
# -CONFIG_USB_OTG_SUPPORTED=y
CONFIG_TINYUSB_CDC_ENABLED=y
CONFIG_TINYUSB_CDC_RX_BUFSIZE=256
CONFIG_TINYUSB_CDC_TX_BUFSIZE=256
CONFIG_USB_AND_UART=y
# -CONFIG_ESP_CONSOLE_USB_CDC=y <--- [no REPL!!]
```

You might want to try my compiled [firmware](https://github.com/shariltumin/others-micropython/tree/master/esp32-s2/Wemos-Mini). The firmware is for the purple S2 Mini V1.0.0 WEMOS.CC board. Give it a try. It might work for you.

**Reply by kjm1102** [reply]
Thnx for the offer. Sadly I've never been able to get esptool running as a stand alone, do all my flashing via Thonny & I don't know how to get it to do shell scripts.
